libraries/mbed/targets/hal/TARGET_NORDIC/TARGET_MCU_NRF51822/sleep.c

```diff
@@ -28,5 +28,5 @@ void sleep(void)
 void deepsleep(void)
 {
     sleep();
-    //   NRF_POWER->SYSTEMOFF=1;
+    NRF_POWER->SYSTEMOFF = 1;
```

Comment:
I did not take into account softdevices when I tested it. However I think that what you are suggesting should be more adequate in the ```sleep(void)``` function. I think that ```deepsleep(void)``` should turn off even the softdevices.